### PR TITLE
adding LightGBMRanker

### PR DESCRIPTION
**Pull request by imatiach-msft**
resolves https://github.com/Azure/mmlspark/issues/543 and https://github.com/Azure/mmlspark/issues/446
TODO: Add tests once new lightgbm native jar uploaded

### CODE REVIEW

**Comment by imatiach-msft** [comment]
Note - this PR requires corresponding changes in LightGBM repository:
https://github.com/Microsoft/LightGBM/pull/2118

File: src/lightgbm/src/main/scala/LightGBMBase.scala [hunk 1]
```diff
@@ -64,15 +64,26 @@ trait LightGBMBase[TrainedModel <: Model[TrainedModel]] extends Estimator[Traine
...
       .reduce((booster1, _) => booster1)
     // Wait for future to complete (should be done by now)
     Await.result(future, Duration(getTimeout, SECONDS))
     getModel(trainParams, lightGBMBooster)
   }
 
+  /** Optional group column for Ranking, set to None by default.
+    * @return None
+    */
+  protected def getOptGroupCol: Option[String] = None
```
Comment (mhamilton723) [inline src/lightgbm/src/main/scala/LightGBMBase.scala]: WHat is this func for?maybe you are looking for `get(groupCol)`?

File: src/lightgbm/src/main/scala/TrainUtils.scala [hunk 2]
```diff
@@ -35,8 +36,27 @@ private object TrainUtils extends Serializable {
     }
   }
 
+  private def addIntField(field: Array[Int], fieldName: String, numRows: Int,
```
Comment (mhamilton723) [inline src/lightgbm/src/main/scala/TrainUtils.scala]: Does this need to be here with the PR that just got merged? Hoping we can remove some of the SWIGTYPE_p_blahs now

File: src/lightgbm/src/main/scala/LightGBMBase.scala (hunk 1 above)
Comment (imatiach-msft) [reply src/lightgbm/src/main/scala/LightGBMBase.scala]: it's just the group column that I pass to map partitions and is required for lightgbm ranker but not needed for classifier or regressor

File: src/lightgbm/src/main/scala/TrainUtils.scala (hunk 2 above)
Comment (imatiach-msft) [reply src/lightgbm/src/main/scala/TrainUtils.scala]: done!  moved to datasets file when rebasing to latest

**Comment by tyler-romero** [comment]
This looks great! Are there plans to add pyspark bindings for LightGBMRanker as well?

**Comment by imatiach-msft** [comment]
@tyler-romero mmlspark autogenerates pyspark and sparklyr bindings for any estimators and transformers added in scala, so the pyspark bindings already exist with this PR :)

File: tools/config.sh [hunk 3]
```diff
@@ -91,9 +91,9 @@ INSTALLATIONS=(
   vers:   "PYTHONDONTWRITEBYTECODE=true conda --version|conda <{ver}>"
   where:  "devel runtime build"
 
-  DataSets ver: "2017-05-25"
-  url:    "$INSTALLER_URL/datasets-<{ver}>.tgz"
-  sha256: "9cf58c6d22fa3d3507608c5af23eb791e37bea324d2c98209ae7356becd4ce41"
+  DataSets ver: "2019-05-02"
+  url:    "https://publictestdatasets.blob.core.windows.net/data/datasets-<{ver}>.tgz"
```
Comment (mhamilton723) [inline tools/config.sh]:
```suggestion
  url:    "$INSTALLER_URL/datasets-<{ver}>.tgz"
```

**Pull request merged**